### PR TITLE
Require SystemCommand only where needed

### PR DESCRIPTION
**Pull request by dduugg**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
Continuation of https://github.com/Homebrew/brew/pull/16538

This removes `SystemCommand::Mixin` from the top-level namespace, and instead includes it only in the needed namespaces.

### CODE REVIEW

File: Library/Homebrew/extend/time.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # typed: true
 # frozen_string_literal: true
 
+require "time"
```
Comment (dduugg) [inline Library/Homebrew/extend/time.rb:4]: This is needed to alias `xmlschema` below (in contexts where it has not already been included, such as `SystemCommand` in this new layout).

**Comment by dduugg** [comment]
https://github.com/Homebrew/homebrew-test-bot/pull/1004 is an attempt to fix the failing check

**Comment by apainintheneck** [comment]
It looks like this is also used in Casks which seems to be covered here and in two cask repo commands as well.

- [changed_files.rb](https://github.com/Homebrew/homebrew-cask/blob/fb9cc3d4e7083471a99ae4808684d09458c9bcba/cmd/lib/changed_files.rb#L5-L15)
- [check.rb](https://github.com/Homebrew/homebrew-cask/blob/fb9cc3d4e7083471a99ae4808684d09458c9bcba/cmd/lib/check.rb#L16)
  - It's used in a few places throughout this file but I just highlighted the first one.

@bevanjkay @reitermarkus How hard would it be to replace these uses of `system_command` with other methods?

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by dduugg** [comment]
> It looks like this is also used in Casks which seems to be covered here and in two cask repo commands as well.
> 
> * [changed_files.rb](https://github.com/Homebrew/homebrew-cask/blob/fb9cc3d4e7083471a99ae4808684d09458c9bcba/cmd/lib/changed_files.rb#L5-L15)
> * [check.rb](https://github.com/Homebrew/homebrew-cask/blob/fb9cc3d4e7083471a99ae4808684d09458c9bcba/cmd/lib/check.rb#L16)
>   
>   * It's used in a few places throughout this file but I just highlighted the first one.
> 
> @bevanjkay @reitermarkus How hard would it be to replace these uses of `system_command` with other methods?

It looks like we can require `brew` paths in this repo, e.g. https://github.com/Homebrew/homebrew-cask/blob/fb9cc3d4e7083471a99ae4808684d09458c9bcba/cmd/lib/generate-matrix.rb#L4

If so, we should be able to `require "system_command"` and include/extend `SystemCommand::Mixin` where appropriate. Does that seem possible?

**Comment by Bo98** [comment]
> @bevanjkay @reitermarkus How hard would it be to replace these uses of `system_command` with other methods?

They should probably be replaced with public API like `Utils.safe_popen_read`

**Comment by dduugg** [comment]
> > It looks like this is also used in Casks which seems to be covered here and in two cask repo commands as well.
> > 
> > * [changed_files.rb](https://github.com/Homebrew/homebrew-cask/blob/fb9cc3d4e7083471a99ae4808684d09458c9bcba/cmd/lib/changed_files.rb#L5-L15)
> > * [check.rb](https://github.com/Homebrew/homebrew-cask/blob/fb9cc3d4e7083471a99ae4808684d09458c9bcba/cmd/lib/check.rb#L16)
> >   
> >   * It's used in a few places throughout this file but I just highlighted the first one.
> > 
> > @bevanjkay @reitermarkus How hard would it be to replace these uses of `system_command` with other methods?
> 
> It looks like we can require `brew` paths in this repo, e.g. https://github.com/Homebrew/homebrew-cask/blob/fb9cc3d4e7083471a99ae4808684d09458c9bcba/cmd/lib/generate-matrix.rb#L4
> 
> If so, we should be able to `require "system_command"` and include/extend `SystemCommand::Mixin` where appropriate. Does that seem possible?

I went ahead and opened a PR to do just this: https://github.com/Homebrew/homebrew-cask/pull/165732

**Pull request merged**